datafusion/sql/src/unparser/expr.rs

```diff
@@ -513,20 +513,30 @@ impl Unparser<'_> {
     fn convert_bound(
         &self,
         bound: &datafusion_expr::window_frame::WindowFrameBound,
-    ) -> ast::WindowFrameBound {
+    ) -> Result<ast::WindowFrameBound> {
         match bound {
             datafusion_expr::window_frame::WindowFrameBound::Preceding(val) => {
-                ast::WindowFrameBound::Preceding(
-                    self.scalar_to_sql(val).map(Box::new).ok(),
-                )
+                Ok(ast::WindowFrameBound::Preceding({
+                    let val = self.scalar_to_sql(val)?;
```

Comment:
There is a subtle difference in how datafusion plans a window frame bound that is `None` vs ScalarValue::Null.

The former yields `PRECEDING UNBOUNDED` and the latter yields `PRECEDING NULL`.

Datafusion's planner accepts the former, but rejects the latter.